client/app/components/calendar/event/EventDestroyer.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { PropTypes } from 'react';
+import { If, Else } from 'react-if';
+import bindAll from 'lodash/bindAll';
+import DeleteButton from 'components/calendar/event/DeleteButton';
+import DeleteTooltip from 'components/calendar/event/DeleteTooltip';
+import DeleteConfirmation from 'components/calendar/event/DeleteConfirmation';
+import './event.scss';
+
+
+export default class EventDestroyer extends React.Component {
+  static propTypes = {
+    onDelete: PropTypes.func.isRequired,
+    disabled: PropTypes.bool.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showIndicator: false,
+      showConfirmationModal: false
+    };
+
+    bindAll(this, ['_handleOnClick', '_handleConfirmDeletion', '_hideConfirmationModal']);
+  }
+
+  render() {
+    return (
+      <div>
+        <DeleteButton onClick={this._handleOnClick} disabled={this.props.disabled} ref="target" />
+        <If condition={this.props.disabled}>
+          <DeleteTooltip show={this.state.showIndicator} target={() => this.refs.target} />
+          <Else>
+            <DeleteConfirmation show={this.state.showConfirmationModal}
+                                cancelHandler={this._hideConfirmationModal}
+                                confirmHandler={this._handleConfirmDeletion}
+                                onHide={this._hideConfirmationModal} />
+          </Else>
+        </If>
+      </div>
+    );
+  }
+
+  _hideConfirmationModal() {
+    this.setState({ showConfirmationModal: false });
+  }
+
+  _handleConfirmDeletion() {
+    this._hideConfirmationModal();
+    return this.props.onDelete();
+  }
+
+  _handleOnClickEnabled() {
+    this.setState({ showConfirmationModal: true });
+  }
+
+  _handleOnClickDisabled() {
+    this.setState({
+      showIndicator: true
```

Comment:
I'm just curious - why here you split the line and in `_handleOnClickEnabled` you didn't.